File: Easy/1-two-sum/try-5.cpp
/*
 * Problem #1: Two Sum
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: cpp
 * Date: 23/11/2024, 13:19:08
 * Link: https://leetcode.com/problems/two-sum/
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int > d;
        for(int i=0;i<nums.size();i++){    
            if(d.find(target-nums[i])!=d.end()){
                return {d[target-nums[i]],i};
            }
            d[nums[i]]=i;
        }
        return {};
    }
};

File: Easy/349-intersection-of-two-arrays/try-5.py
/*
 * Problem #349: Intersection of Two Arrays
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: python3
 * Date: 06/12/2024, 15:02:10
 * Link: https://leetcode.com/problems/intersection-of-two-arrays/
 */

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d={}
        re=[]
        for i in nums1:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        for i in nums2:
            if i in d and d[i]!=0:
                re.append(i)
                d[i]=0
        return re

        
